styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/TerminationHandler.java

```diff
@@ -73,15 +75,28 @@ private void checkRetry(RunState state) {
     final WorkflowInstance workflowInstance = state.workflowInstance();
 
     if (state.data().retryCost() < MAX_RETRY_COST) {
-      final long delayMillis;
-      if (state.data().lastExit().map(c -> c == MISSING_DEPS_EXIT_CODE).orElse(false)) {
-        delayMillis = Duration.ofMinutes(MISSING_DEPS_RETRY_DELAY_MINUTES).toMillis();
+      final Optional<Integer> exitCode = state.data().lastExit();
+      if (shouldFailFast(exitCode)) {
+        stateManager.receiveIgnoreClosed(Event.stop(workflowInstance));
       } else {
-        delayMillis = retryUtil.calculateDelay(state.data().consecutiveFailures()).toMillis();
+        final long delayMillis;
+        if (shouldRetry(exitCode)) {
+          delayMillis = Duration.ofMinutes(MISSING_DEPS_RETRY_DELAY_MINUTES).toMillis();
+        } else {
+          delayMillis = retryUtil.calculateDelay(state.data().consecutiveFailures()).toMillis();
```

Comment:
if shouldRetry == false we... retry? This is a bit confusing. Maybe `shouldRetry` should be named something else?